### PR TITLE
Handle file paths safely for pip

### PR DESCRIPTION
**Pull request by chmeliik**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] New code has type annotations
- [ ] ~Docs updated (if applicable)~
- [ ] ~Docs links in the code are still valid (if docs were updated)~

### CODE REVIEW

File: cachi2/core/package_managers/pip.py [hunk 1]
```diff
@@ -792,7 +781,7 @@ class PipRequirementsFile:
     def __init__(self, file_path):
         """Initialize a PipRequirementsFile.
 
-        :param str file_path: the full path to the requirements file
+        :param file_path: the full path to the requirements file
```
Comment (chmeliik) [inline cachi2/core/package_managers/pip.py:784]: Apologies for not adding type hints to the PipRequirement and PipRequirementsFile classes. Those two (and any methods that use them) cannot possibly pass type-checking without a larger refactoring :cry:

**Comment by chmeliik** [comment]
PTAL, this is mostly just a mechanical replacement of `Path -> RootedPath` + the corresponding changes

File: cachi2/core/package_managers/pip.py (hunk 1 above)
Comment (chmeliik) [reply cachi2/core/package_managers/pip.py:784]: Just to illustrate some of the problems:


PipRequirementsFile initialized with file_path=None:

https://github.com/containerbuildsystem/cachi2/blob/36d398e1d57819d819fd051ee78f611e628f4bb9/cachi2/core/package_managers/pip.py#L808

PipRequirementsFile._read_lines doesn't work if file path is None:

https://github.com/containerbuildsystem/cachi2/blob/36d398e1d57819d819fd051ee78f611e628f4bb9/cachi2/core/package_managers/pip.py#L872

Everything in PipRequirement initialized as None:

https://github.com/containerbuildsystem/cachi2/blob/36d398e1d57819d819fd051ee78f611e628f4bb9/cachi2/core/package_managers/pip.py#L981-L999

None of the PipRequirement methods work if those attributes are None.

File: cachi2/core/package_managers/pip.py [hunk 2]
```diff
@@ -1496,8 +1487,8 @@ def _download_pypi_package(requirement, pip_deps_dir, pypi_url, pypi_auth=None):
     Does not download any dependencies (implied: ignores extras). Ignores environment
     markers (target environment is not known to Cachi2).
 
-    :param PipRequirement requirement: PyPI requirement from a requirement.txt file
-    :param Path pip_deps_dir: The deps/pip directory in a Cachi2 request bundle
+    :param requirement: PyPI requirement from a requirement.txt file
```
Comment (brunoapimentel) [inline cachi2/core/package_managers/pip.py:1490]: Should the type annotation be removed from here (since we're not adding it to the function signature)?

**Comment by brunoapimentel** [comment]
Coverage is down -0.008%, which is fine to me :D

File: cachi2/core/package_managers/pip.py (hunk 2 above)
Comment (chmeliik) [reply cachi2/core/package_managers/pip.py]: Good catch, fixed here and in PipRequirementsFile

**Review by lkolacek — APPROVED** [review]
Looks great! :+1: Thank you Adam

**Pull request merged**